relay/prover.go

```diff
@@ -324,122 +321,17 @@ func (pr *Prover) buildNextSyncCommitteeUpdateForNext(period uint64, trustedHeig
-func (pr *Prover) QueryConnectionWithProof(ctx core.QueryContext) (*conntypes.QueryConnectionResponse, error) {
-	res, err := pr.chain.QueryConnection(ctx)
-	if err != nil {
-		return nil, err
-	}
-	if res.Connection.State == conntypes.UNINITIALIZED {
-		// connection not found
-		return res, nil
-	}
-	proofHeight := int64(ctx.Height().GetRevisionHeight())
-	res.Proof, err = pr.buildStateProof(host.ConnectionKey(
-		pr.chain.Path().ConnectionID,
-	), proofHeight)
-	if err != nil {
-		return nil, err
-	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
-}
-
-// QueryChannelWithProof returns the Channel and its proof
-func (pr *Prover) QueryChannelWithProof(ctx core.QueryContext) (chanRes *chantypes.QueryChannelResponse, err error) {
-	res, err := pr.chain.QueryChannel(ctx)
-	if err != nil {
-		return nil, err
-	}
-	if res.Channel.State == chantypes.UNINITIALIZED {
-		// channel not found
-		return res, nil
-	}
-	proofHeight := int64(ctx.Height().GetRevisionHeight())
-	res.Proof, err = pr.buildStateProof(host.ChannelKey(
-		pr.chain.Path().PortID,
-		pr.chain.Path().ChannelID,
-	), proofHeight)
-	if err != nil {
-		return nil, err
-	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
-}
-
-// QueryPacketCommitmentWithProof returns the packet commitment and its proof
-func (pr *Prover) QueryPacketCommitmentWithProof(ctx core.QueryContext, seq uint64) (comRes *chantypes.QueryPacketCommitmentResponse, err error) {
-	res, err := pr.chain.QueryPacketCommitment(ctx, seq)
-	if err != nil {
-		return nil, err
-	}
-	proofHeight := int64(ctx.Height().GetRevisionHeight())
-	res.Proof, err = pr.buildStateProof(host.PacketCommitmentKey(
-		pr.chain.Path().PortID,
-		pr.chain.Path().ChannelID,
-		seq,
-	), proofHeight)
-	if err != nil {
-		return nil, err
-	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
-}
+var _ core.Prover = (*Prover)(nil)
 
-// QueryPacketAcknowledgementCommitmentWithProof returns the packet acknowledgement commitment and its proof
-func (pr *Prover) QueryPacketAcknowledgementCommitmentWithProof(ctx core.QueryContext, seq uint64) (ackRes *chantypes.QueryPacketAcknowledgementResponse, err error) {
-	res, err := pr.chain.QueryPacketAcknowledgementCommitment(ctx, seq)
-	if err != nil {
-		return nil, err
-	}
+// ProveState returns the proof of an IBC state specified by `path` and `value`
+func (pr *Prover) ProveState(ctx core.QueryContext, path string, value []byte) ([]byte, clienttypes.Height, error) {
 	proofHeight := int64(ctx.Height().GetRevisionHeight())
-	res.Proof, err = pr.buildStateProof(host.PacketAcknowledgementKey(
-		pr.chain.Path().PortID,
-		pr.chain.Path().ChannelID,
-		seq,
-	), proofHeight)
+	height := pr.newHeight(proofHeight)
+	proof, err := pr.buildStateProof([]byte(path), proofHeight)
 	if err != nil {
-		return nil, err
+		return nil, height, err
 	}
-	res.ProofHeight = pr.newHeight(proofHeight)
-	return res, nil
+	return proof, height, nil
```

Comment:
You can simply rewrite this part to one line `return proof, height, err`

Reply:
Remove check err
https://github.com/datachainlab/ethereum-ibc-relay-prover/pull/3/commits/7484802a5ca3f65de988f15af8808db21dce92be